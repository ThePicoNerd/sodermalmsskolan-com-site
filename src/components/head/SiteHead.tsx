import Head from "next/head";
import React from "react";
import ANALYTICS_ID from "../../lib/analytics/constants";
import { MetaHead, SiteMetadata } from "./MetaHead";

export interface SiteHeadProps {
  metadata?: SiteMetadata;
}

/**
 * `<head>` including metadata and icons.
 *
 * @param {React.PropsWithChildren<SiteHeadProps>} props Props.
 *
 * @returns {React.ReactElement} Lots of meta tags.
 */
const SiteHead: React.FunctionComponent<SiteHeadProps> = ({ metadata = {} }) => (
  <>
    <MetaHead {...metadata} />

    <Head>
      <link
        rel="apple-touch-icon"
        sizes="180x180"
        href="/apple-touch-icon.png"
      />
      <link
        rel="icon"
        type="image/png"
        sizes="32x32"
        href="/favicon-32x32.png"
      />
      <link
        rel="icon"
        type="image/png"
        sizes="16x16"
        href="/favicon-16x16.png"
      />
      <link rel="manifest" href="/site.webmanifest" />
      <link rel="mask-icon" href="/safari-pinned-tab.svg" color="#000000" />
      <meta name="msapplication-TileColor" content="#da532c" />
      <meta name="theme-color" content="#ffffff" />

      <script
        async
        src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"
      />

      {/* Global site tag (gtag.js) - Google Analytics */}
      <script
        async
        src={`https://www.googletagmanager.com/gtag/js?id=${ANALYTICS_ID}`}
      />
      <script
        // eslint-disable-next-line react/no-danger
        dangerouslySetInnerHTML={{
          __html: `window.dataLayer = window.dataLayer || [];
              function gtag(){dataLayer.push(arguments);}
              gtag('js', new Date());
              gtag('config', '${ANALYTICS_ID}');`,
        }}
      />
    </Head>
  </>
);

export default SiteHead;
