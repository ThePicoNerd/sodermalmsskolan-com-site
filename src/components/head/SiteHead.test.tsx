import { render } from "@testing-library/react";
import React from "react";
import SiteHead from "./SiteHead";

describe("<SiteHead /> test", () => {
  it("should render correctly", () => {
    const result = render(<SiteHead />);

    expect(result.asFragment()).toMatchSnapshot();
  });
});
