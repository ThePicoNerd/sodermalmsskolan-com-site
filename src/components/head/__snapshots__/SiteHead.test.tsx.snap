// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SiteHead /> test should render correctly 1`] = `
<DocumentFragment>
  <title>
    södermalmsskolan.com
  </title>
  <meta
    content="Vi visar den verkliga innebörden av Sodexos slogan Quality of Life Services."
    name="description"
  />
  <link
    href="https://xn--sdermalmsskolan-8sb.com/"
    rel="canonical"
  />
  <meta
    content="södermalmsskolan.com"
    property="og:site_name"
  />
  <meta
    content="södermalmsskolan.com"
    property="og:title"
  />
  <meta
    content="Vi visar den verkliga innebörden av Sodexos slogan Quality of Life Services."
    property="og:description"
  />
  <meta
    content="https://xn--sdermalmsskolan-8sb.com/"
    property="og:url"
  />
  <meta
    content="website"
    property="og:type"
  />
  <meta
    content="https://cdn.discordapp.com/attachments/575993879837409290/588012243745243136/IMG_20190611_161511.jpg"
    property="og:image"
  />
  <link
    href="/apple-touch-icon.png"
    rel="apple-touch-icon"
    sizes="180x180"
  />
  <link
    href="/favicon-32x32.png"
    rel="icon"
    sizes="32x32"
    type="image/png"
  />
  <link
    href="/favicon-16x16.png"
    rel="icon"
    sizes="16x16"
    type="image/png"
  />
  <link
    href="/site.webmanifest"
    rel="manifest"
  />
  <link
    color="#000000"
    href="/safari-pinned-tab.svg"
    rel="mask-icon"
  />
  <meta
    content="#da532c"
    name="msapplication-TileColor"
  />
  <meta
    content="#ffffff"
    name="theme-color"
  />
  <script
    async=""
    src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"
  />
  <script
    async=""
    src="https://www.googletagmanager.com/gtag/js?id=G-P3DFFXTCY0"
  />
  <script>
    window.dataLayer = window.dataLayer || [];
              function gtag(){dataLayer.push(arguments);}
              gtag('js', new Date());
              gtag('config', 'G-P3DFFXTCY0');
  </script>
</DocumentFragment>
`;
